kernel/private/classes/clusterfilehandlers/dfsbackends/mysqli.php

```diff
@@ -392,11 +391,15 @@ public function _purgeByLike( $like, $onlyExpired = false, $limit = 50, $expiry
             return $this->_fail( "Purging file metadata by like statement $like failed" );
         }
         $deletedDBFiles = mysqli_affected_rows( $this->db );
-        $this->dfsbackend->delete( $files );
 
-        $this->_commit( $fname );
+        if ( $this->_commit( $fname ) === true )
```

Comment:
This is problematic, I know. Aside from the issue of adding a return value to `_commit`, what happens if the commits are recursive and `null` is returned? The files won't be deleted, but if the outer commit eventually goes through, we would want them to be. OTOH if we did delete them in the null case, and the outer commit was rolled back, we'd have the opposite problem.